spec/factories/journeys.rb

```diff
@@ -0,0 +1,12 @@
+FactoryBot.define do
+  factory :journey do
+    association(:move)
+    association(:supplier)
+    association(:from_location, factory: :location)
+    association(:to_location, :court, factory: :location)
+    client_timestamp { Time.now.utc + rand(-60..60).seconds } # NB: the client_timestamp will never be perfectly in sync with system clock
+
+    # NB we need to initialize_state because FactoryBot fires the after_initialize callback before the attributes are initialised!
+    after(:build) { |object| object.send(:initialize_state) }
```

Comment:
I'm wondering if there's a less jiggery pokery way of achieving the correct call order of callbacks 🤔

Reply:
I couldn't see a better way - I need to set the state for `new()`, for `build()`, for `create()` and for `find()`... Open to other suggestions though?

Reply:
I'd need to read through how they've plumbed together finite_state_machine to understand this. You've got it working so I'm fine with this for the meanwhile